src/streaming/controllers/MediaController.js

```diff
@@ -310,46 +315,62 @@ function MediaController() {
         return notEmpty ? settings : null;
     }
 
-    function matchSettings(settings, track, isTrackActive = false) {
-        try {
-            let matchLang = false;
-
-            // If there is no language defined in the target settings we got a match
-            if (!settings.lang) {
-                matchLang = true;
+    function filterTracksBySettings(tracks, filterFn, settings) {
+        let tracksAfterMatcher = [];
+        tracks.forEach(function (track) {
+            if (filterFn(settings, track)) {
+                tracksAfterMatcher.push(track);
             }
+        });
+        if (tracksAfterMatcher.length !== 0) {
+            return tracksAfterMatcher;
+        }
+        return tracks;
```

Comment:
In case we have no match should we return an empty array? Or is the idea to ignore this specific filter then because it was called with an invalid or non existent value?

Reply:
As discussed verbally, we return the array of matches only if it is not empty; otherwise we keep the list of tracks unchanged (and subsequently preform a selection based on `selectionModeForInitialTrack`).